ZFSin/zfs/module/zfs/vdev_disk.c

```diff
@@ -549,61 +608,60 @@ vdev_disk_io_start(zio_t *zio)
 
 	zio->io_target_timestamp = zio_handle_io_delay(zio);
 
-	/*
-	* If we use vdev_disk_io_intr() as the IoSetCompletionRoutine() we BSOD as
-	* the IoSetCompletionRoutine() is called in higher priority, and vdev_disk_io_intr()
-	* calls zio_taskq_dispatch() which uses mutex calls, and that is not allowed at
-	* that IRQ level. So for now we block waiting on IoSetCompletionRoutine() setting
-	* and Event, then we manually call vdev_disk_io_intr().
-	* We should change this to call zio_taskq_dispatch() before IO, but to a new
-	* taskq, which immediately blocks waiting for Event to be set. That way we
-	* as async, and not blocking.
-	*/
-
 	ASSERT(zio->io_size != 0);
 
-	NTSTATUS status;
 	PIRP irp = NULL;
 	PIO_STACK_LOCATION irpStack = NULL;
-	void *b_addr = NULL;
 	IO_STATUS_BLOCK IoStatusBlock = { 0 };
 	LARGE_INTEGER offset;
-	KEVENT Event;
 
 	offset.QuadPart = zio->io_offset + vd->vdev_win_offset;
 
+	/* Preallocate space for IoWorkItem, required for vdev_disk_io_start_done callback */
+	vd_callback_t *vb = (vd_callback_t *)kmem_alloc(sizeof(vd_callback_t) + IoSizeofWorkItem(), KM_SLEEP);
+
+	if (!vb) {
```

Comment:
No need, if you call `kmem_alloc( ..., KM_SLEEP)` it will never return NULL, but sleep until there is memory.

Reply:
ok, will remove.